CJBItemSpawner/Framework/ItemMenu.cs

```diff
@@ -267,15 +270,25 @@ public override void receiveRightClick(int x, int y, bool playSound = true)
         public override void receiveKeyPress(Keys key)
         {
             bool inDropdown = this.CategoryDropdown.IsExpanded;
+            bool closeMenuButton = key == Keys.Escape || Game1.options.doesInputListContain(Game1.options.menuButton, key)
+                || Game1.options.doesInputListContain(Game1.options.cancelButton, key);
 
-            // deselect textbox
-            if (this.SearchBox.Selected && key == Keys.Escape)
-                this.DeselectSearchBox();
-
+            if (this.SearchBox.Selected && closeMenuButton)
+            {
+                // cleaer textbox
+                if (!string.IsNullOrWhiteSpace(this.SearchBox.Text) && this.SearchBox.Text.Length > 0)
```

Comment:
No need to check the length, since we already checked it has a non-whitespace character:
```cs
if (!string.IsNullOrWhiteSpace(this.SearchBox.Text))
```